### PR TITLE
Fix convertJson configuration parsing

### PR DESCRIPTION
**Pull request by rozza**
SPARK-425

### CODE REVIEW

File: src/main/java/com/mongodb/spark/sql/connector/config/WriteConfig.java [hunk 1]
```diff
@@ -87,16 +86,28 @@ public enum ConvertJson {
       this.value = operationType;
     }
 
-    static ConvertJson fromString(final String jsonType) {
-      if (jsonType.equalsIgnoreCase(TRUE)) {
+    static ConvertJson fromString(final String operationType) {
```
Comment (stIncMale) [inline src/main/java/com/mongodb/spark/sql/connector/config/WriteConfig.java:89]: [optional]
Not that I understand how the previous name "jsonType" was a good description for a value of the `convertJson` option, but neither do I understand why "operationType" is a good description for that. Wouldn't `operationType` or even just `v` be more relevant?

File: src/test/java/com/mongodb/spark/sql/connector/config/MongoConfigTest.java [hunk 2]
```diff
@@ -127,6 +127,22 @@ void testMongoConfigWriteConcern() {
     assertEquals(new WriteConcern("region1"), writeConfig.getWriteConcern());
   }
 
+  @Test
+  void testWriteConfigConvertJson() {
+    WriteConfig writeConfig = MongoConfig.createConfig(CONFIG_MAP).toWriteConfig();
+    assertEquals(writeConfig.convertJson(), WriteConfig.ConvertJson.FALSE);
+    assertEquals(
+        writeConfig.withOption("convertJson", "True").convertJson(), WriteConfig.ConvertJson.ANY);
```
Comment (stIncMale) [inline src/test/java/com/mongodb/spark/sql/connector/config/MongoConfigTest.java:135]: [optional]
Let's also add
```java
assertEquals(
        writeConfig.withOption("convertJson", "False").convertJson(), WriteConfig.ConvertJson.FALSE);
```

**Review by stIncMale — APPROVED** [review]
There are two optional items. Feel free to merge with or without addressing them.

File: src/main/java/com/mongodb/spark/sql/connector/config/WriteConfig.java (hunk 1 above)
Comment (rozza) [reply src/main/java/com/mongodb/spark/sql/connector/config/WriteConfig.java:89]: I just used it as its the same name in the constructor.  Will update.

File: src/test/java/com/mongodb/spark/sql/connector/config/MongoConfigTest.java (hunk 2 above)
Comment (rozza) [reply src/test/java/com/mongodb/spark/sql/connector/config/MongoConfigTest.java:138]: Done

**Pull request merged**